modules/selection/src/main/kotlin/com.r3.corda.lib.tokens.selection/memory/services/VaultWatcherService.kt

```diff
@@ -117,42 +129,45 @@ class VaultWatcherService(tokenObserver: TokenObserver? = null) : SingletonSeria
     @Suspendable
     fun lockTokensExternal(list: List<StateAndRef<FungibleToken>>, knownSelectionId: String) {
         list.forEach {
-            val token = it.state.data
-            val (owner, type, typeId) = processToken(token)
-            val tokensForTypeInfo = getTokenBucket(owner, type, typeId)
+            val idx = processToken(it)
+            val tokensForTypeInfo = getTokenBucket(idx)
             tokensForTypeInfo.replace(it, PLACE_HOLDER, knownSelectionId)
         }
     }
 
     @Suspendable
     fun selectTokens(
-            owner: Any,
-            amountRequested: Amount<IssuedTokenType>,
+            owner: Holder,
```

Comment:
if the indexing method is set to Token only then does the `Holder.MappedIdentity` choice do anything?

Reply:
no, this is for indexing by external id. You can have keys related to external id (`MappedIdentity`), and you can have ones that are `Unmapped`. for selection just by token you have `TokenOnly`

Reply:
If you want I can revert to `Any` but I introduced type safety because I was asked to do that.

Reply:
I don't follow - let's chat about it after lunch!